drivers/i2c/i2c_dw.c

```diff
@@ -54,8 +55,12 @@ static inline void i2c_dw_data_ask(const struct device *dev)
 		return;
 	}
 
+	/* Get the FIFO depth that could be from 2 to 256 from HW spec */
+	rx_buffer_depth = regs->ic_comp_param_1.bits.rx_buffer_depth;
+	tx_buffer_depth = regs->ic_comp_param_1.bits.tx_buffer_depth;
+
 	/* How many bytes we can actually ask */
-	rx_empty = (I2C_DW_FIFO_DEPTH - read_rxflr(reg_base)) - dw->rx_pending;
+	rx_empty = (rx_buffer_depth - regs->ic_rxflr) - dw->rx_pending;
```

Comment:
Looks like this is out of date; reference to `regs` is now replaced with functions involving `reg_base`.

Reply:
Thank you. I'm trying to correct that now. Just a moment, please.

Reply:
I fixed the issue.